src/Microsoft.Health.Dicom.Core.UnitTests/Features/Store/Entries/DicomInstanceEntryReaderForSinglePartRequestTests.cs

```diff
@@ -80,5 +87,56 @@ public async Task GivenBodyPartWithValidContentType_WhenReading_ThenCorrectResul
                     Assert.Same(_stream, await item.GetStreamAsync(source.Token));
                 });
         }
+
+        [Fact]
+        public async Task GivenBodyPartWithValidContentTypeExceedLimit_ThrowsError()
```

Comment:
Can you add one more test where the data is exactly the max length?

Reply:
Added